### PR TITLE
Adding test suite documentation.

### LINKED ISSUE #134: Documentation update on automated testing with go
Provide instructions on environment setup and building unit tests / integration tests using terratest for Bedrock

### PR DESCRIPTION
**Pull request by NathanielRose**
Closes #134

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+In addition, the bedrock test suite allows for better collaboration with embedding into CI/CD tools such as Travis or Azure DevOps Pipelines.
+
+## Prerequisites
+
+Please [install bedrock required tools](/cluster/README.md/#required-tools) in addition to the following:
+
+- [Go](https://golang.org/doc/install)
+- [Dep](https://github.com/golang/go/wiki/PackageManagementTools)
+- [Terratest Modules](https://github.com/gruntwork-io/terratest)
```
Comment (dtzar) [inline test/README.md]: remove - this will get installed automatically below with the dep command

File: test/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+- CI/CD Tool (Optional)
```
Comment (dtzar) [inline test/README.md]: Not necessary, I'd remove

File: test/README.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+## Test Setup Locally
+
+In this example we are using the [`azure-simple`](/cluster/environments/azure-simple/readme.md) for a template integration test.
```
Comment (dtzar) [inline test/README.md]: Add explicit instructions on having this repo exist in the user's `GOPATH`

Comment (NathanielRose) [reply test/README.md]: Added some notes to maintain repo in src directory.

**Comment by dtzar** [comment]
As more tests are added, we'll need to specify a way to run only specific tests or test suites.  Looks good now!

**Pull request merged**